### DISCUSSION TITLE
Welcome to Duende.IdentityServer.Admin Discussions!

### DISCUSSION THREAD

**Discussion by skoruba** [skoruba/Duende.IdentityServer.Admin#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by prabakardotnet** [comment]
Hi, could we use / tweak this UI to work with openiddict library?

**Reply by skoruba** [reply]
Hello, unfortunately not, openiddict has different API.

Thanks

**Comment by vzam** [comment]
Hi, I want to use the Admin UI in a project but there are some obstacles which I am wondering about, maybe you can clarify the reasoning for them:

- Why is the business logic in a NuGet package? Especially that logic will have to be changed sooner or later for a project, and actually I have already the need to do that in the very early stages of my project
- What is the benefit of using the template instead of just forking the project? It makes merging features a little bit harder and removes the ability to change the business logic layers. Unfortunately modifying the business logic in a fork is still hard because the layers heavily use generics, so custom methods and properties of the concrete classes become unavailable there
- Why is the Admin UI not hosted itself but as part of the Admin project? Since ASP.NET is not using the build folder as the base directory when debugging, only the razor views are rendered correctly but the wwwroot stuff is missing altogether, which includes everything from bootstrap. I understand that there is a special configuration in docker compose for debugging but I prefer just starting the debugger in my IDE without having to build an image. Shouldn't the seeding be done by the STS.Identity?
